src/main/java/org/jasig/portlet/calendar/adapter/CalendarEventsDao.java

```diff
@@ -58,240 +58,236 @@
+     *
+     * @param e
+     * @param interval
+     * @param usersConfiguredDateTimeZone
+     * @return
+     * @throws IOException
+     * @throws URISyntaxException
+     * @throws ParseException
+     */
+    protected Set<CalendarDisplayEvent> getDisplayEvents(VEvent e, Interval interval, Locale locale, DateTimeZone usersConfiguredDateTimeZone) throws IOException, URISyntaxException, ParseException {
+
+        final VEvent event = (VEvent) e.copy();
+
+        DateTime eventStart;
+        DateTime eventEnd = null;
+
+        if (event.getStartDate().getTimeZone() == null && !event.getStartDate().isUtc()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Identified event " + event.getSummary() + " as a floating event");
+            }
+
+            int offset = usersConfiguredDateTimeZone.getOffset(event.getStartDate().getDate().getTime());
+            eventStart = new DateTime(event.getStartDate().getDate().getTime()-offset, usersConfiguredDateTimeZone);
+            if (event.getEndDate() != null) {
+                eventEnd = new DateTime(event.getEndDate().getDate().getTime()-offset, usersConfiguredDateTimeZone);
+            }
+
+        } else {
+            eventStart = new DateTime(event.getStartDate().getDate(), usersConfiguredDateTimeZone);
+            if (event.getEndDate() != null) {
+                eventEnd = new DateTime(event.getEndDate().getDate(), usersConfiguredDateTimeZone);
+            }
+        }
+
+        if (eventEnd == null) {
+            eventEnd = eventStart;
+        }
+
+        // Multi-day events may begin in the past;  make sure to choose a date in range for the first pass...
+        final Date firstDayToProcess = interval.contains(event.getStartDate().getDate().getTime())
+                ? event.getStartDate().getDate()
+                : interval.getStart().toDate();
+
+        DateMidnight startOfTheSpecificDay = new DateMidnight(firstDayToProcess, usersConfiguredDateTimeZone);
+        DateMidnight endOfTheSpecificDay = startOfTheSpecificDay.plusDays(1);
+
+        final DateTimeFormatter df = getDateFormatter(locale, usersConfiguredDateTimeZone);
+        final DateTimeFormatter tf = getTimeFormatter(locale, usersConfiguredDateTimeZone);
+        final Set<CalendarDisplayEvent> events = new HashSet<CalendarDisplayEvent>();
+        final Interval eventInterval = new Interval(eventStart, eventEnd);
+
+        do {
+            final Interval theSpecificDay = new Interval(startOfTheSpecificDay.getMillis(), endOfTheSpecificDay.getMillis(), usersConfiguredDateTimeZone);
+
+            /*
+             * The test if the event interval has two parts: one more subtle than the other. The later test of overlaps()
+             * is an obvious test if the event happens in the specific day. The former test handles the special case
+             * where a zero-duration event abuts the specific day. The abuts() method checks to see if the intervals
```

Comment:
I was confused in reading this about the mention of the `abuts()` method, which doesn't seem to be used in the method.  Either this could be clearer that it is drawing a distinction between the logic here and the logic that could have been had invoking an `abuts()` method, or this means to be talking about some other method, or something?